### PR TITLE
actions/docker: add ~el9~,f40,noble drop focal

### PR DESCRIPTION
**Pull request by trws**
problem: we're using old images and out of date WRT core

solution: pull in current versions and update build matrix

### CODE REVIEW

**Comment by trws** [comment]
NOTE: Dropping EL9 from this so it can proceed separately from the matrix repair upstream.  Will add that back in once flux-framwork/flux-core#6156 lands

File: src/test/docker/fedora40/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,36 @@
+FROM fluxrm/flux-core:fedora40
+
+LABEL maintainer="Mark A. Grondona <mark.grondona@gmail.com>"
```
Comment (jameshcorbett) [inline src/test/docker/fedora40/Dockerfile:3]: Is this right?

**Review by jameshcorbett — APPROVED** [review]
Looks reasonable, one question. He's listed as the maintainer on the other fedora dockerfiles, just checking that he's ok with continuing to be listed there.

File: src/test/docker/fedora40/Dockerfile (hunk 1 above)
Comment (trws) [reply src/test/docker/fedora40/Dockerfile:3]: I can change it at least here, good point.

**Pull request merged**